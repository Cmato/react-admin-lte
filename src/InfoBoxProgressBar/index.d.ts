export { default } from './InfoBoxProgressBar';
export * from './InfoBoxProgressBar';
