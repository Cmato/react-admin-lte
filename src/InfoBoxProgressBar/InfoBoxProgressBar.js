// @inheritedComponent ProgressBar

import React from 'react';
import PropTypes from 'prop-types';
import ProgressBar from 'react-bootstrap/lib/ProgressBar';

const InfoBoxProgressBar = ({
  description,
  ...other
}) => (
  <div>
    <ProgressBar key="bar" {...other} />,
    <span key="description" className="progress-description">{description}</span>
  </div>
);

InfoBoxProgressBar.propTypes = {
  /**
   * Description of progress.
   */
  description: PropTypes.node.isRequired
};

export default InfoBoxProgressBar;
