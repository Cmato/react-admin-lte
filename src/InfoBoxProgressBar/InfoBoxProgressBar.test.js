import React from 'react';
import { shallow } from 'enzyme';
import InfoBoxProgressBar from './InfoBoxProgressBar';

it('renders with props passed to bar', () => {
  expect(shallow((
    <InfoBoxProgressBar
      now={67}
      description="i made progress"
    />
  ))).toMatchSnapshot();
});
