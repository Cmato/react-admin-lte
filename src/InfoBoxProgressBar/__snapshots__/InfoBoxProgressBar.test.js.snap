// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props passed to bar 1`] = `
<div>
  <ProgressBar
    active={false}
    bsClass="progress-bar"
    isChild={false}
    key="bar"
    max={100}
    min={0}
    now={67}
    srOnly={false}
    striped={false}
  />
  <span
    className="progress-description"
    key="description"
  >
    i made progress
  </span>
</div>
`;
